File: Code blocks/array_with_odd_sum.c
#include<stdio.h>
int main()
{
    int t,n,c,a,sum[2000],i,x,z[2000],j,y,zz[2000];
    scanf("%d",&t);
    for(i=0;i<t;i++)
    {
        c=0;
        x=0;
        y=0;
        scanf("%d",&n);
        for(j=0;j<n;j++)
        {
            scanf("%d",&a);
            c=c+a;
            if(a%2==1)
            {
                x++;
            }
            else
            {
                y++;
            }
        }
        sum[i]=c;
        z[i]=x;
        zz[i]=y;
    }
    for(i=0;i<t;i++)
    {
        if(sum[i]%2==0)
        {
            if(zz[i]==0)
            {
                printf("NO\n");
            }
            else if(z[i]==0)
            {
                printf("NO\n");
            }
            else
            {
                printf("YES\n");
            }
        }
        else
        {
            printf("YES\n");
        }
    }
    return 0;
}

File: Python/array_with_odd_sum.py
import sys
input=sys.stdin.buffer.readline


t=int(input())
while t:
	t-=1
	n=int(input())
	arr=list(map(int,input().split()))
	if sum(arr)%2==1:
		print("YES")
	else:
		odd=0
		even=0
		for i in range(n):
			if arr[i]%2==1:
				odd+=1
			if arr[i]%2==0:
				even+=1
			if even>0 and odd>0:
				break
		if even>0 and odd>0:
			print("YES")
		else:
			print("NO")


